Mathport/Syntax/Translate/Tactic/Lean3.lean

```diff
@@ -300,51 +298,51 @@ where
   | AST3.Expr.const ⟨_, `ff⟩ _ _, a, f => f a false
   | _, a, _ => a
 
-def quoteSimpConfig (cfg : Meta.Simp.Config) : Option Syntax := Id.run do
+def quoteSimpConfig (cfg : Meta.Simp.Config) : Option Term := Id.run do
   if cfg == {} then return none
   let a := #[]
-    |> push cfg {} qnat `maxSteps (·.maxSteps)
-    |> push cfg {} qnat `maxDischargeDepth (·.maxDischargeDepth)
-    |> push cfg {} qbool `contextual (·.contextual)
-    |> push cfg {} qbool `memoize (·.memoize)
-    |> push cfg {} qbool `singlePass (·.singlePass)
-    |> push cfg {} qbool `zeta (·.zeta)
-    |> push cfg {} qbool `beta (·.beta)
-    |> push cfg {} qbool `eta (·.eta)
-    |> push cfg {} qbool `iota (·.iota)
-    |> push cfg {} qbool `proj (·.proj)
-    |> push cfg {} qbool `decide (·.decide)
+    |> push cfg {} `maxSteps (·.maxSteps)
+    |> push cfg {} `maxDischargeDepth (·.maxDischargeDepth)
+    |> push cfg {} `contextual (·.contextual)
+    |> push cfg {} `memoize (·.memoize)
+    |> push cfg {} `singlePass (·.singlePass)
+    |> push cfg {} `zeta (·.zeta)
+    |> push cfg {} `beta (·.beta)
+    |> push cfg {} `eta (·.eta)
+    |> push cfg {} `iota (·.iota)
+    |> push cfg {} `proj (·.proj)
+    |> push cfg {} `decide (·.decide)
   `({ $[$a:structInstField],* })
 where
-  qbool (b : Bool) : Syntax := mkIdent $ cond b `true `false
-  qnat : ℕ → Syntax := Quote.quote
-  push {β} [BEq β] {α} (a b : α) (q : β → Syntax)
-    (n : Name) (f : α → β) (args : Array Syntax) : Array Syntax := Id.run do
+  push {β} [BEq β] {α} (a b : α) [Quote β]
```

Comment:
IIRC the reason for the `q` function is because the `Quote Bool` instance fully qualifies `true` and `false` but we are trying to generate the unqualified form here.

Reply:
Ah, I was wondering why we had a custom quote implementation here.